### PR TITLE
added basic epa-ng recipe

### PR DESCRIPTION
**Pull request by gavinmdouglas**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

File: recipes/epa-ng/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package:
+  name: epa-ng
+  version: "0.3.3"
+
+build:
+  number: 1
```
Comment (epruesse) [inline recipes/epa-ng/meta.yaml]: new recipe, should be 0

File: recipes/epa-ng/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  build:
+    - {{ compiler('cxx') }}
+    - libtool
+    - flex
+    - bison
+    - m4
+    - cmake==3.11
+    - automake
+    - autoconf
+  run:
```
Comment (epruesse) [inline recipes/epa-ng/meta.yaml]: zlib should be a host dependency. The run dependency will be injected automatically.

File: recipes/epa-ng/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    - zlib
+
+test:
+  commands:
+    - epa-ng -h > /dev/null
+
+about:
+  home: https://github.com/Pbdas/epa-ng
+  license: GNU Affero General Public License v3.0
```
Comment (epruesse) [inline recipes/epa-ng/meta.yaml]: GPL requires that the license be distributed with the packages. Please add `license_file:`

File: recipes/epa-ng/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  summary: Massively parallel phylogenetic placement of genetic sequences
+
+extra:
+  identifiers:
+    - https://doi.org/10.1093/sysbio/syy054
```
Comment (epruesse) [inline recipes/epa-ng/meta.yaml]: should be `- doi:10.1093/sysbio/syy054`

File: recipes/epa-ng/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
+package:
+  name: epa-ng
+  version: "0.3.3"
+
+build:
+  number: 1
+
+source:
+  url: https://github.com/Pbdas/epa-ng/archive/v0.3.3.tar.gz
+  md5: 881d024877f3cfa632d344373e119f9d
```
Comment (epruesse) [inline recipes/epa-ng/meta.yaml]: we prefer sha256.

File: recipes/epa-ng/conda_build_config.yaml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+c_compiler:
+  - gcc      # [linux]
+  - clang    # [osx]
+cxx_compiler:
+  - gxx      # [linux]
+  - clangxx  # [osx]
```
Comment (epruesse) [inline recipes/epa-ng/conda_build_config.yaml]: I presume it doesn't work without this file?

File: recipes/epa-ng/meta.yaml (hunk 2 above)
Comment (epruesse) [reply recipes/epa-ng/meta.yaml]: The build/host/run is a little confusing. It's for cross compiling. Everything that is build machinery, so compiler and build tools go into `build`. Everything that is dependent on the build target, so libraries you need to link against, goes into `host`. Things that will be needed at run time go in to `run`. Some packages have what conda calls `run exports`. Those will add things to your `run` automatically. Zlib is one of them. The `run: - zlib` is therefore redundant.

Comment (gavinmdouglas) [reply recipes/epa-ng/meta.yaml]: Yes I was confused by this; however, without adding zlib as a run requirement this error is causes: ```SERR epa-ng: error while loading shared libraries: libz.so.1: cannot open shared object file: No such file or directory```, which you can see here: https://circleci.com/gh/bioconda/bioconda-recipes/38521?utm_campaign=workflow-failed&utm_medium=email&utm_source=notification

File: recipes/epa-ng/meta.yaml (hunk 5 above)
Comment (gavinmdouglas) [reply recipes/epa-ng/meta.yaml]: Changed.

File: recipes/epa-ng/conda_build_config.yaml (hunk 6 above)
Comment (gavinmdouglas) [reply recipes/epa-ng/conda_build_config.yaml]: Yes this resulted from extensive troubleshooting, which you can read about here: https://forum.qiime2.org/t/issue-with-cmake-when-making-epa-ng-conda-recipe/6430/9

File: recipes/epa-ng/meta.yaml (hunk 3 above)
Comment (epruesse) [reply recipes/epa-ng/meta.yaml]: Yes. It needs to be distributed with the binaries as well. By specifying `license_file: LICENSE` conda-build knows which file to add to the package (ends up in the `info` folder).

File: recipes/epa-ng/meta.yaml (hunk 2 above)
Comment (epruesse) [reply recipes/epa-ng/meta.yaml]: It needs to be in `host`.

Comment (epruesse) [reply recipes/epa-ng/meta.yaml]: I added that - should work.

File: recipes/epa-ng/meta.yaml (hunk 3 above)
Comment (gavinmdouglas) [reply recipes/epa-ng/meta.yaml]: Ok thanks - I added it

**Comment by gavinmdouglas** [comment]
Hi @epruesse, it looks like all of the minor edits have been addressed.

**Comment by gavinmdouglas** [comment]
Thanks for your help @epruesse.

**Pull request merged**